src/main/java/liquibase/ext/hibernate/snapshot/SequenceSnapshotGenerator.java

```diff
@@ -34,6 +34,12 @@ protected void addTo(DatabaseObject foundObject, DatabaseSnapshot snapshot) thro
             Schema schema = (Schema) foundObject;
             HibernateDatabase database = (HibernateDatabase) snapshot.getDatabase();
             for (org.hibernate.boot.model.relational.Namespace namespace : database.getMetadata().getDatabase().getNamespaces()) {
+                boolean namespaceMatchesSchema = (namespace.getName().getSchema() != null && namespace.getName().getSchema().matches(foundObject.getName()))
```

Comment:
This check is duplicated in `SequenceSnapshotGenerator` and `TableSnapshotGenerator`. Could you extract it to a generic method in `HibernateSnapshotGenerator` , or even move the check there so it's not repeatable? As I believe it would be necessary to `IndexSnapshotGenerator` and other `SnapshotGenerators` too.. or am I overreacting?

Reply:
Sure, I can extract the check. I don't think we need to move the check though, since from my understanding, `DatabaseObjects` other than `Schema`, `Table` and `Sequence` are added only from the corresponding snapshot-generators. 
So I think if we have the check in those three generators, we don't need the check in the other ones, since the database objects that get to them always already have the correct schema. 
I might be wrong though, I don't fully understand the snapshotting mechanism.

Reply:
done